hack/cmd_util_test.sh

```diff
@@ -0,0 +1,337 @@
+
+set -o errexit
+set -o nounset
+set -o pipefail
+# set -x
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${OS_ROOT}/hack/util.sh"
+source "${OS_ROOT}/hack/cmd_util.sh"
+os::log::install_errexit
+
+mkdir -p /tmp/openshift/origin/test/cmd/
+JUNIT_OUTPUT_FILE=/tmp/openshift-cmd/junit_output.txt
+
+# set verbosity so we can see that command output renders correctly
+VERBOSE=1
+
+# positive tests
+os::cmd::expect_success 'exit 0'
+
+os::cmd::expect_failure 'exit 10'
+
+os::cmd::expect_success_and_text 'printf "hello" && exit 0' 'hello'
+
+os::cmd::expect_failure_and_text 'printf "hello" && exit 19' 'hello'
+
+os::cmd::expect_success_and_not_text 'echo "goodbye" && exit 0' 'hello'
+
+os::cmd::expect_failure_and_not_text 'echo "goodbye" && exit 19' 'hello'
+
+os::cmd::expect_code 'exit 195' '195'
+
+os::cmd::expect_code_and_text 'echo "hello" && exit 213' '213' 'hello'
+
+os::cmd::expect_code_and_not_text 'echo "goodbye" && exit 213' '213' 'hello'
+
+echo "positive tests: ok"
+
+# negative tests
+
+if os::cmd::expect_success 'exit 1'; then
+	exit 1
+fi
+
+if os::cmd::expect_failure 'exit 0'; then
+	exit 1
+fi
+
+if os::cmd::expect_success_and_text 'echo "goodbye" && exit 0' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_success_and_text 'echo "hello" && exit 1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_success_and_text 'echo "goodbye" && exit 1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_failure_and_text 'echo "goodbye" && exit 1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_failure_and_text 'echo "hello" && exit 0' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_failure_and_text 'echo "goodbye" && exit 0' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_success_and_not_text 'echo "hello" && exit 0' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_success_and_not_text 'echo "goodbye" && exit 1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_success_and_not_text 'echo "hello" && exit 1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_failure_and_not_text 'echo "goodbye" && exit 0' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_failure_and_not_text 'echo "hello" && exit 1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_failure_and_not_text 'echo "hello" && exit 0' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_code 'exit 1' '200'; then
+	exit 1
+fi
+
+if os::cmd::expect_code_and_text 'echo "hello" && exit 0' '1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_code_and_text 'echo "goodbye" && exit 1' '1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_code_and_text 'echo "goodbye" && exit 0' '1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_code_and_not_text 'echo "goodbye" && exit 0' '1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_code_and_not_text 'echo "hello" && exit 1' '1' 'hello'; then
+	exit 1
+fi
+
+if os::cmd::expect_code_and_not_text 'echo "hello" && exit 0' '1' 'hello'; then
+	exit 1
+fi
+
+echo "negative tests: ok"
+
+# complex input tests
```

Comment:
@liggitt also specifically if you can think of other "complex" input that should be tested, I've put my best guess at a covering set here